src/Api/PubnubApi/Security/Crypto/Cryptors/CryptorBase.cs

```diff
@@ -0,0 +1,106 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Security.Cryptography;
+using System.Text;
+using PubnubApi.Security.Crypto.Common;
+
+namespace PubnubApi.Security.Crypto.Cryptors
+{
+    public abstract class CryptorBase : ICryptor
+    {
+        protected const int IV_SIZE = 16;
+
+        private readonly bool _useDynamicRandomIV;
+        private readonly IPubnubLog _log;
+        private readonly IPubnubUnitTest _unitTest;
+        protected CryptorBase(string cipherKey, bool useDynamicRandomIV, IPubnubLog log, IPubnubUnitTest unitTest)
```

Comment:
Let's not pass UnitTests in constructors - we can invert it by exposing event delegates

Reply:
the only purpose is pass known IV for encryption, so that we can verify the expected output. Any other option?

Reply:
removed UnitTests from constructors.